tests/conftest.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# MeshPy: A beam finite element input generator
+#
+# MIT License
+#
+# Copyright (c) 2018-2024
+#     Ivo Steinbrecher
+#     Institute for Mathematics and Computer-Based Simulation
+#     Universitaet der Bundeswehr Muenchen
+#     https://www.unibw.de/imcs-en
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+# -----------------------------------------------------------------------------
+"""Testing framework infrastructure."""
+
+import os
+from pathlib import Path
+
+import pytest
+from _pytest.config import Config
+from _pytest.config.argparsing import Parser
+
+
+def pytest_addoption(parser: Parser) -> None:
+    """Add custom command line options to pytest.
+
+    Args:
+        parser: Pytest parser
+    """
+
+    parser.addoption(
+        "--4C_ArborX",
+        action="store_true",
+        default=False,
+        help="Execute 4C and ArborX based tests.",
+    )
+
+    parser.addoption(
+        "--CubitPy",
+        action="store_true",
+        default=False,
+        help="Execute CubitPy based tests.",
+    )
+
+    parser.addoption(
+        "--performance",
+        action="store_true",
+        default=False,
+        help="Execute performance tests.",
+    )
+
+
+def pytest_collection_modifyitems(config: Config, items: list) -> None:
+    """Filter tests based on their markers and provided command line options.
+
+    Currently configured options:
+        `pytest`: execute tests with no markers
+        `pytest --4C_ArborX`: execute tests with the `fourc_arborx` marker
```

Comment:
If I read this code correctly, we always execute not tagged tests, right? Then this has to be adapted.

In general I really like the current approach (and kudos to the implementation!) of always executing the base tests and then allowing to selectively add test tags. Two comments from my side:
1. I would split 4C and ArborX, as a few people use MeshPy directly with 4C but don't have ArborX configured.
1. When running the performance tests, I would like to not run the untagged tests, is this possible?

Reply:
Good point - I'll split the 4C and ArborX tests

We could simply add a flag if the base tests are executed? Then this could be simply switched on and off for the performance tests as well all other tests

Reply:
Added an additional flag `--exclude-standard-tests` and it works as expected

`pytest` executes all standard tests
`pytest --4C` executes all standard tests + 4C tests
`pytest --4C --exclude-standard-tests` executes only the 4C tests

4C can be swapped with all other flags or an arbitrary number of flags can be combined, i.e.,

`pytest --4C --ArborX --CubitPy --Performance --exclude-standard-tests` executes only the special tests, i.e. 4C, ArborX, CubitPy and Performance test

Really nice solution in my opinion! Thanks for your input @isteinbrecher